src/libraries/Microsoft.PowerFx.Core/Utils/Validation.cs

```diff
@@ -280,6 +310,7 @@ public static void CheckNull<T>(T? val, string paramName)
             }
         }
 
+        [Conditional("DEBUG")]
```

Comment:
The point of the `Check` methods is that they are _**NOT**_ debug only, and should only be used in code that needs this validation in release builds. 
`Assert` is debug-only.